cs/HomeExercises/NumberValidatorTests.cs

```diff
@@ -5,76 +5,140 @@
+        }
+
+        [TestCase(4, 2, true, "a.sd", TestName = "IsValidNumber Invalid Case: Value Is Not a Number")]
+        [TestCase(4, 2, true, "0.a", TestName = "IsValidNumber Invalid Case: Value Is Not a Number")]
+        [TestCase(4, 2, true, "a.0", TestName = "IsValidNumber Invalid Case: Value Is Not a Number")]
+        [TestCase(4, 2, true, " ", TestName = "IsValidNumber Invalid Case: Value Is Not a Number")]
+        public void TestIsValidNumberValueIsNotNumber(int precision, int scale, bool onlyPositive, string value)
+        {
+            var numberValidator = new NumberValidator(precision, scale, onlyPositive);
+            numberValidator.IsValidNumber(value).Should().BeFalse($"{value} should not be a number");
+        }
+
+        [TestCase(4, 2, true, "12345.12", TestName = "IsValidNumber Invalid Case: Value Length Greater Precision")]
+        public void TestIsValidNumberValueLengthGreaterPrecision(int precision, int scale, bool onlyPositive, string value)
+        {
+            var numberValidator = new NumberValidator(precision, scale, onlyPositive);
+            numberValidator.IsValidNumber(value).Should().BeFalse($"{value} length should be greater {precision}");
+        }
+
+        [TestCase(4, 2, true, "1.123", TestName = "IsValidNumber Invalid Case: Frac Length Greater Scale")]
+        public void TestIsValidNumberFracLengthGreaterScale(int precision, int scale, bool onlyPositive, string value)
+        {
+            var numberValidator = new NumberValidator(precision, scale, onlyPositive);
+            numberValidator.IsValidNumber(value).Should().BeFalse($"{value} frac part should be greater {scale}");
+        }
+
+        [TestCase(4, 2, true, "-1.1", TestName = "IsValidNumber Invalid Case: Value Is Negative While OnlyPositive Is True")]
+        public void TestIsValidNumberValueIsNegativeWhileOnlyPositiveIsTrue(int precision, int scale, bool onlyPositive, string value)
+        {
+            var numberValidator = new NumberValidator(precision, scale, onlyPositive);
+            numberValidator.IsValidNumber(value).Should().BeFalse($"{value} should be negative while OnlyPositive is {onlyPositive}");
+        }
+
+        [TestCase(4, 2, true, "12.12", TestName = "IsValidNumber: Valid Cases")]
+        [TestCase(4, 2, true, "1.1", TestName = "IsValidNumber: Valid Cases")]
+        [TestCase(4, 2, true, "1,1", TestName = "IsValidNumber: Valid Cases")]
+        [TestCase(4, 2, false, "12.12", TestName = "IsValidNumber: Valid Cases")]
+        [TestCase(4, 2, false, "-1.12", TestName = "IsValidNumber: Valid Cases")]
+        public void TestIsValidNumberValidCases(int precision, int scale, bool onlyPositive, string value)
+        {
+            var numberValidator = new NumberValidator(precision, scale, onlyPositive);
+            numberValidator.IsValidNumber(value).Should().BeTrue($"{value} should be a valid number");
+        }
+    }
```

Comment:
Я бы все-таки пообъединяла некоторые тесткейсы (все валидные с валидными, невалидные с невалидными) где это возможно, так будет гораздо проще читать тесты. Для этого придется либо убрать because (и я бы посоветовала этот вариант, у нас в команде он крайне редко используется, когда сценарий действительно не совсем понятный), либо вынести их в параметры тестов и заполнять в тесткейсах (но не вижу в этом особого смысла, так как сценарии простые)

Reply:
объединил